test/pardot-test-srv/main.go

```diff
@@ -0,0 +1,256 @@
+		"token_type":   "Bearer",
+		"expires_in":   3600,
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	err = json.NewEncoder(w).Encode(response)
+	if err != nil {
+		log.Printf("Failed to encode token response: %v", err)
+		http.Error(w, "Failed to encode token response", http.StatusInternalServerError)
+	}
+}
+
+func (ts *testServer) createContactsHandler(w http.ResponseWriter, r *http.Request) {
+	ts.token.Lock()
+	validToken := ts.token.active
+	ts.token.Unlock()
+
+	token := r.Header.Get("Authorization")
+	businessUnitId := r.Header.Get("Pardot-Business-Unit-Id")
+
+	if token != "Bearer "+validToken {
+		http.Error(w, "Unauthorized", http.StatusUnauthorized)
+		return
+	}
+
+	if businessUnitId == "" {
+		http.Error(w, "Missing 'Pardot-Business-Unit-Id' header", http.StatusBadRequest)
+		return
+	}
+
+	body, err := io.ReadAll(r.Body)
+	if err != nil {
+		http.Error(w, "Failed to read request body", http.StatusInternalServerError)
+		return
+	}
+
+	type contactData struct {
+		Email string `json:"email"`
+	}
+
+	var contact contactData
+	err = json.Unmarshal(body, &contact)
+	if err != nil {
+		http.Error(w, "Failed to parse request body", http.StatusBadRequest)
+		return
+	}
+
+	if contact.Email == "" {
+		http.Error(w, "Missing 'email' field in request body", http.StatusBadRequest)
+		return
+	}
+
+	if ts.developmentMode {
+		ts.contacts.Lock()
+		_, exists := ts.contacts.created[contact.Email]
+		if !exists {
+			ts.contacts.created[contact.Email] = struct{}{}
+		}
+		ts.contacts.Unlock()
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	err = json.NewEncoder(w).Encode(map[string]string{"status": "success"})
+	if err != nil {
+		http.Error(w, "Failed to encode response", http.StatusInternalServerError)
+		return
+	}
+}
+
+func (ts *testServer) queryContactsHandler(w http.ResponseWriter, r *http.Request) {
+	ts.contacts.Lock()
+	respContacts := make([]string, 0, len(ts.contacts.created))
+	for contact := range ts.contacts.created {
+		respContacts = append(respContacts, contact)
+	}
+	ts.contacts.Unlock()
+
+	w.Header().Set("Content-Type", "application/json")
+	err := json.NewEncoder(w).Encode(map[string]interface{}{"contacts": respContacts})
+	if err != nil {
+		log.Printf("Failed to encode contacts query response: %v", err)
+		http.Error(w, "Failed to encode contacts query response", http.StatusInternalServerError)
+	}
+}
+
+func main() {
+	oauthAddr := flag.String("oauth-addr", "", "OAuth server listen address override")
+	pardotAddr := flag.String("pardot-addr", "", "Pardot server listen address override")
+	configFile := flag.String("config", "", "Path to configuration file")
+	flag.Parse()
+
+	if *configFile == "" {
+		flag.Usage()
+		os.Exit(1)
+	}
+
+	file, err := os.Open(*configFile)
+	cmd.FailOnError(err, "Opening configuration file")
+	defer file.Close()
+	decoder := json.NewDecoder(file)
+	var c config
+	err = decoder.Decode(&c)
+	cmd.FailOnError(err, "Unmarshalling configuration file")
```

Comment:
Since we already import `cmd` in this file, `cmd.ReadConfigFile` would be a convenient shorthand. And gets us "reject unknown fields" for free.

Reply:
I could have sworn I had removed this as well, perhaps I lost a changeset when I moved some things around for the final PRs.